### PR TITLE
Feature/78 Support for Exasol 7.0.0

### LINKED ISSUE #78: Add Support for Exasol 7.0

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #78

### CODE REVIEW

File: src/main/java/com/exasol/containers/ExasolContainer.java [hunk 1]
```diff
@@ -55,7 +56,13 @@
      * @see ExasolDockerImageReference#parse(String) Examples for supported reference types
      */
     public ExasolContainer(final String dockerImageName) {
-        super(ExasolDockerImageReference.parse(dockerImageName).toString());
+        this(ExasolDockerImageReference.parse(dockerImageName));
+
```
Comment (morazow) [inline src/main/java/com/exasol/containers/ExasolContainer.java:60]:
```suggestion
```

File: src/main/java/com/exasol/containers/ExasolContainer.java [hunk 2]
```diff
@@ -98,10 +105,49 @@ protected void configure() {
         super.configure();
     }
 
+    /**
+     * Get the default internal port of the database.
+     * <p>
+     * This method dispatches between the port numbers depending on the version of the Exasol database. This is
```
Comment (morazow) [inline src/main/java/com/exasol/containers/ExasolContainer.java:111]:
```suggestion
     * This method chooses the port number depending on the version of the Exasol database. This is
```

File: src/main/java/com/exasol/containers/ExasolContainer.java [hunk 3]
```diff
@@ -98,10 +105,49 @@ protected void configure() {
...
+            return DEFAULT_CONTAINER_INTERNAL_DATABASE_PORT_V7_AND_ABOVE;
+        } else {
+            return DEFAULT_CONTAINER_INTERNAL_DATABASE_PORT;
+        }
+    }
+
+    /**
+     * Get the default internal port of the BucketFS.
+     * <p>
+     * This method dispatches between the port numbers depending on the version of the Exasol database. This is
```
Comment (morazow) [inline src/main/java/com/exasol/containers/ExasolContainer.java:131]:
```suggestion
     * This method chooses the BucketFS port number depending on the version of the Exasol database. This is
```

File: src/main/java/com/exasol/containers/ExasolDockerImageReference.java [hunk 4]
```diff
@@ -59,4 +52,32 @@ public static ExasolDockerImageReference parse(final String reference) {
             return new ExasolDockerImageReference(reference);
         }
     }
+
+    /**
+     * Get the major version of the {@code exasol/docker-db} image if possible.
+     * 
+     * <p>
+     * If a different image is used, not version is detected.
```
Comment (morazow) [inline src/main/java/com/exasol/containers/ExasolDockerImageReference.java:60]:
```suggestion
     * If a different image is used, no version is detected.
```

File: src/main/java/com/exasol/containers/ExasolDockerImageReference.java [hunk 5]
```diff
@@ -12,22 +13,14 @@
 public class ExasolDockerImageReference {
     private static final Pattern DOCKER_IMAGE_VERSION_PATTERN = Pattern
             .compile("(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:-d(\\d+))?");
+    private static final Pattern MAJOR_VERSION_PATTERN = Pattern
+            .compile(Pattern.quote(EXASOL_DOCKER_IMAGE_ID) + ":(\\d+).*");
```
Comment (morazow) [inline src/main/java/com/exasol/containers/ExasolDockerImageReference.java:17]: Here, I think you can use the previous DOCKER_IMAGE_VERSION_PATTERN pattern. From looking into the pattern, you should be able to get the major version. In the getMajorVersion method it gets the first matched group from the version.

Could you please check it, if you have not already?

File: src/test/java/com/exasol/containers/ExasolDockerImageReferenceTest.java [hunk 6]
```diff
@@ -19,4 +19,14 @@
     void toString(final String input, final String expectedReference) {
         assertThat(ExasolDockerImageReference.parse(input).toString(), equalTo(expectedReference));
     }
+
+    @CsvSource({ //
+            "7, 7", //
+            "6.1, 6", //
+            "exasol/docker-db:6.2.7-d1, 6" //
```
Comment (morazow) [inline src/test/java/com/exasol/containers/ExasolDockerImageReferenceTest.java:26]:
```suggestion
            "exasol/docker-db:6.2.7-d1, 6", //
            "exasol/docker-db:7.0.1, 7" //
```

**Review by morazow — CHANGES_REQUESTED** [review]

File: src/main/java/com/exasol/containers/ExasolDockerImageReference.java (hunk 5 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/containers/ExasolDockerImageReference.java:17]: I just checked it again. the difference is that `MAJOR_VERSION_PATTERN` only matches on the official docker db. For that I added the `Pattern.quote(EXASOL_DOCKER_IMAGE_ID)` which renders to `exasol/docker-db`.

Comment (morazow) [reply src/main/java/com/exasol/containers/ExasolDockerImageReference.java:17]: Okay, makes sense. So, not to match on custom images.

**Review by morazow — APPROVED** [review]

File: src/main/java/com/exasol/containers/ExasolDockerImageReference.java (hunk 5 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/containers/ExasolDockerImageReference.java:17]: :+1:

**Pull request merged**